### ISSUE TITLE
macos -- auth0 opens finder rather than browser for authentication

### ISSUE THREAD

**Issue by dangogh** [auth0/auth0-cli#354]
**Please do not report security vulnerabilities here**. The [Responsible Disclosure Program](https://auth0.com/whitehat) details the procedure for disclosing security issues.

**Thank you in advance for helping us to improve this library!** Please read through the template below and answer all relevant questions. Your additional work here is greatly appreciated and will help us respond as quickly as possible. For general support or usage questions, use the [Auth0 Community](https://community.auth0.com/) or [Auth0 Support](https://support.auth0.com/). Finally, to avoid duplicates, please search existing Issues before submitting one here.

By submitting an Issue to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

> Provide a clear and concise description of the issue, including what you expected to happen.

On MacOS when attempting to login, auth0 cli opens the macos Finder app rather than opening a browser,  so login is not possible.

### Reproduction

Installed auth0 cli from  `auth0 login` on terminal command line using `brew` instructions from README.md
On command line:  `auth0 login` and got the following output:
```
✪ Welcome to the Auth0 CLI 🎊

If you don't have an account, please go to https://auth0.com/signup

Your Device Confirmation code is:

 ▸    Press Enter to open the browser to log in or ^C to quit...

Waiting for login to complete in browser... ⣷ Waiting for login to complete in browser... failed

=== ffn-dev.us.auth0.com error

 ▸    login error: Unauthorized
```

I expected to have a browser window open in Chrome to login to auth0, but instead the Finder app opened to my current directory, so could not complete the login process.

### Environment

- **Version of this library used:**
   - `auth0 version 0.10.3 0f57d1303347dd7bd7b8547ed81026b46b98089c`
- **Version of the platform or framework used, if applicable:**
  - macos 11.5.2
- **Other relevant versions (language, server software, OS, browser):**
  - chrome `Version 93.0.4577.63 (Official Build) (x86_64)

**Comment by Widcket** [comment]
Hi @dangogh, thanks for raising this.
I just tried reproducing, but was unsuccessful:

<img width="712" alt="Screen Shot 2021-09-15 at 18 31 50" src="https://user-images.githubusercontent.com/5055789/133512817-078cedb2-29fd-45b4-8958-bd439af488c2.png">

The latest version of the CLI uses pkg/browser to handle opening URLs across platforms. On macOS, that package calls `open` on the URL: https://github.com/pkg/browser/blob/master/browser_darwin.go

Running, for example, `open https://github.com/pkg/browser/blob/master/browser_darwin.go` manually on the Terminal does open it in the browser. Perhaps you have another protocol helper set up in `~/Library/Preferences/com.apple.LaunchServices/com.apple.LaunchServices.plist`?

**Comment by dangogh** [comment]
That's what's surprising..   I use `open` on the cmd line quite a lot as well as cmd-click on a url in the iterm2 to open in the browser.   Just to double-check, I ran the open you suggested -- opened in chrome as expected...  and I am running the latest version: ```auth0 --version                                                                                    [15:40:33]

auth0 version 0.10.3 0f57d1303347dd7bd7b8547ed81026b46b98089c```

**Comment by dangogh** [comment]
fyi -- this is on a corporate-controlled macbook, so there are things on it I can't reconfigure or remove..

**Comment by dangogh** [comment]
More info:   I just compiled myself (go 1.16.7) and that version worked as expected..

**Comment by Widcket** [comment]
I just tried the released version and was unable to reproduce either.

<img width="666" alt="Screen Shot 2021-09-16 at 23 55 27" src="https://user-images.githubusercontent.com/5055789/133716897-ff3f63ee-9edd-47a6-94b2-51ed73eb7f01.png">

I'll close this issue, as seems to be something related to a particular local environment.

**Comment by dangogh** [comment]
@Widcket I found the issue -- I had `AUTH0_CLIENT_ID` set in my terminal from some other work I was doing.  Without that, it opens the URL in the browser as it should.  Perhaps a check for that with an appropriate error message?

**Comment by dangogh** [comment]
Just tried 0.11 as well -- same behavior when `AUTH0_CLIENT_ID` is set.

**Comment by Widcket** [comment]
Just tried running `AUTH0_CLIENT_ID=xxxxxxxxxxxx auth0 test login` and it opened the URL in the browser.

**Comment by btilford** [comment]
Unsetting `AUTH0_CLIENT_ID` made this work for me.

**Comment by dangogh** [comment]
@btilford glad I wasn't the only one :-)

**Comment by andrewclaus** [comment]
I have the same issue. If AUTH0_CLIENT_ID is set, it redirects me to finder. However I cannot remove this because I am using a personalized cloud, and the readme says to set this value

**Comment by Widcket** [comment]
As a workaround, with the latest version (0.11.5) you can pass `--no-input` to the login command, and it will print the login URL instead of trying to open it in the browser. Then you can copy and paste the URL in the browser and continue the login normally. See https://github.com/auth0/auth0-cli/pull/390

**Comment by andrewclaus** [comment]
@Widcket Thanks for the quick response. Unfortunately that command did not change the output. I will open a new issue for my problem as I think it is more involved.

**Comment by jsoref** [comment]
In principle, this was fixed by #470
